app/code/Magento/Wishlist/view/frontend/web/js/add-to-wishlist.js

```diff
@@ -92,19 +96,30 @@ define([
             if (isFileUploaded) {
                 this.bindFormSubmit();
             }
-            this._updateAddToWishlistButton(dataToAdd);
+            this._updateAddToWishlistButton(dataToAdd, sku);
             event.stopPropagation();
         },
 
         /**
          * @param {Object} dataToAdd
+         * @param {String} sku
          * @private
          */
-        _updateAddToWishlistButton: function (dataToAdd) {
-            var self = this;
+        _updateAddToWishlistButton: function (dataToAdd, sku) {
+            var parentClass = '',
+                addToWishlistBtn = '',
+                params = '',
+                self = this;
 
-            $('[data-action="add-to-wishlist"]').each(function (index, element) {
-                var params = $(element).data('post');
+            if (sku !== '') {
+                parentClass = $('body').hasClass('catalog-product-view') ?
```

Comment:
Please move selectors name to variables and use variables in the code. This will improve code visibility and readability. Thanks.

Reply:
Hello @VladimirZaets ,

I have change selector with variable.
Please review the code.

Thanks
Nirav Patel